### DISCUSSION TITLE
How to get job metadata and run stats

### DISCUSSION THREAD

**Discussion by dagsir** [dagster-io/dagster#8417] [Q&A] [answered]
Is there a way to get job metadata like start time, duration or endtime?

---

*The question was originally asked in [Dagster Slack](https://dagster.slack.com/archives/C01U954MEER/p1641924977185700?thread_ts=1641924977.185700&cid=C01U954MEER).*

**Comment by yuhan** [comment, marked as answer]
In a sensor or other places where the context arg is available, you can get run stats (timestamp, etc) from Dagster Instance by passing in a run_id:
```
@sensor(job=my_job)
def my_sensor(context):
    run_stats = instance.get_run_stats(my_run_id) # run_stats.start_time, run_stats.end_time, ...
```

If you'd also want to get these info outside a sensor, you can do:
```
from dagster import DagsterInstance

instance = DagsterInstance.get()
run_stats = instance.get_run_stats(my_run_id)
```
